Reloop Beatmix (v1) mapping

uncleeugene opened the pull request:
Reloop Beatmix gen.1 mapping added. Documentation is pulled as well.

uncleeugene committed:
Reloop Beatmix gen.1 mapping added.

uncleeugene committed:
Revert "Reloop Beatmix gen.1 mapping added."

This reverts commit 688b5e7aa8c722a6ac6d6cb73511bf383878fe63.

uncleeugene committed:
Reloop Beatmix gen.1 mapping added.

JoergAtGithub commented:
Welcome at Mixxx!
As a first-time contributor we need you to sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and comment here when you have done so. It gives us permission to distribute your contribution under the GPL v2 or later license and the Apple Mac App Store. It is also helpful for us to have contact information for contributors in case we may need it in the future.

ronso0 commented on res/controllers/Reloop_Beatmix.js:
```diff
@@ -0,0 +1,206 @@
+const FULL = 0x7F;
+const OFF = 0x00;
+
+const bNAVKNOB = 0x18;
+const bBEATMASH1 = 0x21;
+const bBEATMASH2 = 0x22;
+const QUICKJOGR = 0x26;
+const bPLUS = 0x27;
+const QUICKJOGL = 0x28;
+const bMINUS = 0x26;
+const bSHIFTMINUS = 0x46;
+const bSHIFTPLUS = 0x47;
+const bSHIFTCUE = 0x4E;
+const bKEYLOCK = 0x4F;
+const bSHIFTLOAD = 0x52;
```
why the `b` prefix and not all capitals for all?

uncleeugene replied:
Fixed.

ronso0 reviewed (changes requested):
Thank you for this mapping!

I just skimmed through an added some first comments.

Please sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and comment here when you have done so. It gives us permission to distribute your contribution under the GPL v2 or later license and the Apple Mac App Store. It is also helpful for us to have contact information for contributors in case we may need it in the future.

uncleeugene committed:
A small change in .desktop file in order to fix gnome dock integration
StartupWMClass added. With this property set now the icon in gnome dock
shown properly and it is now possible to pin mixxx app to dock.

uncleeugene committed:
Revert "A small change in .desktop file in order to fix gnome dock integration"

This reverts commit f1fb0103a9ee1a7b11b04d0c02e99312eac0655f.

uncleeugene committed:
Update res/controllers/Reloop Beatmix.midi.xml

Agree. I took this line from some other script i used as a base.

Co-authored-by: ronso0 <ronso0@mixxx.org>

uncleeugene committed:
Update res/controllers/Reloop Beatmix.midi.xml

Same here, this line is inherited from base script.

Co-authored-by: ronso0 <ronso0@mixxx.org>

uncleeugene committed:
Update res/controllers/Reloop_Beatmix.js

Readability iprovement

Co-authored-by: ronso0 <ronso0@mixxx.org>

uncleeugene commented on res/controllers/Reloop_Beatmix.js:
```diff
@@ -0,0 +1,206 @@
+/****************************************************************
+*       Reloop Beatmix script v0.9                              *
+*       Author: Eugene Erokhin, uncle.eugene@gmail.com          *
+*       Tested on Mixxx version 2.4.x                           *
+*                                                               *
+*       Documentation in the Mixxx wiki:                        *
+*       https://mixxx.org/wiki/doku.php/reloop_beatmix          *
+****************************************************************/
```
Sorry, my bad :)

uncleeugene committed:
Update res/controllers/Reloop_Beatmix.js

Co-authored-by: ronso0 <ronso0@mixxx.org>

uncleeugene commented on res/controllers/Reloop Beatmix.midi.xml:
```diff
@@ -0,0 +1,1266 @@
+            <control>
+                <group>[Channel1]</group>
+                <key>loop_in</key>
+                <description>MIDI Learned from 2 messages.</description>
```
Thought about it, but forgot at the end of the day. Will clean up.

uncleeugene replied:
Fixed.

uncleeugene committed:
Mostly cosmetic changes to improve readability.

uncleeugene commented:
I've signed the agreement.
Git is giving me hard times :) One have to use it for some time to wrap his head around it. I have pushed another commit in that same branch by mistake, but reverted the commit. sorry about that.
Anyways, i updated all that matters with another push. Is that the proper way to update a pull request by just pushing all the changes?

uncleeugene commented on res/controllers/Reloop_Beatmix.js:
```diff
@@ -0,0 +1,206 @@
+// If you want to use fader as a hotcue button then what are you, a monster? :)
+Beatmix.fader = function(_channel, _control, value, _status, group) {
+    const deck = script.deckFromGroup(group);
+    if (Beatmix.faderStartEnabled[deck] && value > faderStartThr && Beatmix.faderLastValue[deck] < faderStartThr) {
```
Fixed.

ronso0 commented:
Thank you.

That's a good moment to learn about `git rebase -i`. I suggest to rebase onto 2.4 and use the `s` option to squash all commits into one.

uncleeugene committed:
One incorrect entry removed from xml file.

uncleeugene commented:
> That's a good moment to learn about `git rebase -i` and `git push -f`. I suggest to rebase onto 2.4 and use the `s` option to squash all commits into one.

I kinda understand what rebase is about but my understanding of the whole structure of git workflow is loose. So i definitely need to start from very basics :)

ronso0 commented on res/controllers/Reloop Beatmix.midi.xml:
```diff
@@ -0,0 +1,1146 @@
+                    <normal/>
+                </options>
+            </control>
+            <control>
+                <group>[Master]</group>
+                <key>headGain</key>
+                <status>0xB4</status>
+                <midino>0x51</midino>
+                <options>
+                    <normal/>
+                </options>
+            </control>
```
Are you sure this does not affect the hardware output? (disable this mapping to double-check)
Just wondering because on my TerminalMix 2 and 4's it does, and they're also from 2012.

uncleeugene replied:
I checked it just now by closing headphone out with hardware pot and opening it back with mouse. Since there's just one headphone volume control i assume it is software :) Master is the only hardware control. Well there's mic gain and jog sensitivity trims as well.

uncleeugene committed:
Filenames changed in order to meet file naming convention.

uncleeugene commented:
> The [file name convention](https://github.com/mixxxdj/mixxx/wiki/Contributing%20mappings#file-naming-convention) is to use hyphens instead of spaces.
> 
> See also #12333
> 
> (I thought I left a comment before but most not have hit enter!)

Done :)

ronso0 commented on res/controllers/Reloop-Beatmix.midi.xml:
```diff
@@ -0,0 +1,1146 @@
+<?xml version='1.0' encoding='utf-8'?>
+<MixxxControllerPreset mixxxVersion="2.0.0+" schemaVersion="1">
+    <info>
+        <name>Reloop Beatmix</name>
```
```suggestion
        <name>Reloop Beatmix Mk1</name>
```
